fuse_optimizers/src/fixed_lag_smoother.cpp

```diff
@@ -426,12 +442,12 @@ void FixedLagSmoother::transactionCallback(
     // The pending set is arranged "smallest stamp last" to making popping off the back more efficient
     auto comparator = [](const ros::Time& value, const TransactionQueueElement& element)
     {
-      return value >= element.stamp();
+      return value >= element.minStamp();
```

Comment:
This change I'm not sure about. I mentioned this a bit in another PR.
If we order the transactions by minStamp(), then the transactions will not be processed in "measurement order".
For example:
IMU measures the state at t1 and the constraint involves only t1.
Relative odometry measures the state at t2 and the constraint involves t0 and t2.
Sorting by min stamp adds the odometry constraint measured at t2 _before_ adding the IMU constraint measured at t1.

To be clear, I have no idea if this is a problem at all. It just _feels_ wrong somehow...to me.

I suppose the one argument I have is from the motion model stand point. By inserting the intermediate states first, the motion model has to do slightly less work.
Example 1:
* Current motion model stamp is t0.
* Process IMU measurement with involved stamps (t1) creates segment t0->t1
* Process the Odometry measurement with involved stamps (t0, t2) creates segment t1->t2

Example 2:
* Current motion model stamp is t0.
* Process the Odometry measurement with involved stamps (t0, t2) creates segment t0->t2
* Process IMU measurement with involved stamps (t1) deletes segment t0->t2, creates segment t0->t1 and t1->t2

The motion model is capable of handling either. And the above example may be contrived and doesn't happen in reality. Perhaps a more realistic scenario is:
* Current motion model stamp is t0.
* Process IMU measurement with involved stamps (t2) creates segment t0->t2
* Process the Odometry measurement with involved stamps (t1, t3) deletes segment t0->t2, creates segment t0->t1, t1->t2, t2->3

In which case, any ordering will involve a lot of deleting and adding.

The `processQueue()` method has also evolved quite a bit from its original design. With the additional of blacklisting transactions from a particular sensor and continuing processing, we no longer add transactions to the graph in a strict order. It may be the actual order of processing is fairly arbitrary at this point, as long as the filtering is in place.

Curious to hear any thoughts you have?

Reply:
> The processQueue() method has also evolved quite a bit from its original design. With the additional of blacklisting transactions from a particular sensor and continuing processing, we no longer add transactions to the graph in a strict order. It may be the actual order of processing is fairly arbitrary at this point, as long as the filtering is in place.

My understanding is that a sensor is blacklisted only if `applyMotionModels` fails, and I think that rarely happens. So, unless I'm wrong, `processQueue()` would process all the transactions and try to apply the motion models iterating the `pending_transactions_` queue in reverse order, that is, from older to newer stamp.

> Curious to hear any thoughts you have?

I also feel there's something that doesn't smell well if we don't use the measurement time, as you've mentioned.

I think the processing order should still use the measurement time, i.e. the transaction `stamp()`, which is typically the `maxStamp()` of the transaction involved stamps.

But for purging, I feel like the`minStamp()` of the transaction involved stamps should also be taken into account.